project-koku/koku
# [COST-3545] Add distributed cost to OpenShift provider map

## Jira Ticket

[COST-3545](https://issues.redhat.com/browse/COST-3545)

## Description

This change will add distributed cost to OCP provider map and show a `distributed` cost field in the cost total for the total and for each project in the api response when grouping by project.

## Testing

1. Checkout Branch
2. Restart Koku
3. load test data
4. Hit endpoint 
    ```
    http://127.0.0.1:8000/api/cost-management/v1/reports/openshift/costs/?group_by[project]=*&group_by[cluster]=Test OCP on Premises
    ```

    1. You should see a `distributed` field in the cost total in the api response.
    
    2. If you search for `Platform unallocated`, the value for distributed cost for those projects should have `0.0` value.
    
    ```
        "distributed": {
            "value": 0.0,
            "units": "USD"
        },
    ```

## Notes

...
